x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/query/AbstractBooleanScriptFieldQuery.java

```diff
@@ -6,32 +6,24 @@
 
 package org.elasticsearch.xpack.runtimefields.query;
 
-import org.apache.lucene.index.LeafReaderContext;
-import org.apache.lucene.search.ConstantScoreScorer;
-import org.apache.lucene.search.ConstantScoreWeight;
-import org.apache.lucene.search.DocIdSetIterator;
-import org.apache.lucene.search.IndexSearcher;
 import org.apache.lucene.search.QueryVisitor;
-import org.apache.lucene.search.ScoreMode;
-import org.apache.lucene.search.Scorer;
-import org.apache.lucene.search.TwoPhaseIterator;
-import org.apache.lucene.search.Weight;
 import org.elasticsearch.script.Script;
 import org.elasticsearch.xpack.runtimefields.BooleanScriptFieldScript;
 import org.elasticsearch.xpack.runtimefields.DoubleScriptFieldScript;
 
-import java.io.IOException;
-import java.util.Objects;
-
 /**
  * Abstract base class for building queries based on {@link DoubleScriptFieldScript}.
  */
-abstract class AbstractBooleanScriptFieldQuery extends AbstractScriptFieldQuery {
-    private final BooleanScriptFieldScript.LeafFactory leafFactory;
+abstract class AbstractBooleanScriptFieldQuery extends AbstractScriptFieldQuery<BooleanScriptFieldScript> {
 
     AbstractBooleanScriptFieldQuery(Script script, BooleanScriptFieldScript.LeafFactory leafFactory, String fieldName) {
-        super(script, fieldName);
-        this.leafFactory = Objects.requireNonNull(leafFactory);
+        super(script, fieldName, leafFactory::newInstance);
+    }
+
+    @Override
+    protected boolean matches(BooleanScriptFieldScript scriptContext, int docId) {
+        scriptContext.runForDoc(docId);
```

Comment:
is scriptContext a good name here? I wish that we rename also the different BooleanScriptFieldScript in a followup. That is effectively the script context for boolean runtime fields?

Reply:
It's really a `script` more than a `scriptContext`. The context is all of the information we need to compile a script.

Reply:
Yeah. That `Script` class is more of a `ScriptConfiguration` or a `ScriptRequest` or something like that. These are the "real" scripts. I think, at least. Because they instances of these object "run".

Reply:
ok I will followup with a proposal to rename these and the corresponding lucene queries